### ISSUE TITLE
"Hello World" Examples dont' Work.

### ISSUE THREAD

**Issue by astorm** [open-telemetry/opentelemetry-php#60]
**Describe your environment** 

The docker-compose.yaml file in this repo, running on a MacOS docker host.  The master branch as of today.

**Steps to reproduce**
1. [Fix a few bugs in the sample code](https://github.com/open-telemetry/opentelemetry-php/pull/59)

2. Changed sample code to use http://127.0.0.1 instead of `zipkin` as hostname (or make a local host entry that points the name `zipkin` to 127.0.0.1)

3. Run `docker-compose up -d` 

3. Run `php ./examples/AlwaysOnTraceExample.php` from your computer -- everything works! Zipkin is running!

3. Run `docker-compose run --rm php php ./examples/AlwaysOnTraceExample.php` or `make examples`, per the README.md

**What is the expected behavior?**

Everything works!  No errors, spans/traces are send to zipkin.  A new user has a good first experience with the repo. 

**What is the actual behavior?**

I receive the following errors

> Warning: file_get_contents(http://127.0.0.1:9411/api/v2/spans): 
> failed to open stream: Connection refused in
> /usr/src/myapp/src/Exporter/ZipkinExporter.php on line 63

I was sad.  I thought hard about my life choices and what I chose to get good at. 

**Additional context**

It seems like the the provided docker-compose command executes PHP _inside_ the container, and the `php` container can't talk to the `zipkin` container. 

I'm a little uncertain on expectations here and/or if I'm missing something, so I can't suggest a solution.  My end goal would be for a new user to the repository to have a successful first run experience when following the provided instructions.

Also -- re: the `zipkin` hostname in the sample code -- should something in docker have set this up?  The README said to use 127.0.0.1 for the hostname.  This is something else that would be great to clear up for new users.

**Comment by astorm** [comment]
@bobstrecansky @beniamin Hello!  I noticed you two showed up on the git ~blame~ show-me-who-did-this list for the docker-composer.yaml file.  If you have a moment, could you comment on the above?  Am I misunderstanding the intent of the php/zipkin containers and how they should interact, or is this a legit bug that should be fixed (and that you'd accept a PR for?)

**Comment by bobstrecansky** [comment]
@astorm - just tried this out myself.  I performed the following steps:

```➜  opentelemetry-php git:(master) ✗ git reset --hard upstream/master
HEAD is now at d8a554b Fixes Broken Example Code and Link in README (#59)
➜  opentelemetry-php git:(master) ✗ docker-compose up -d
Starting opentelemetry-php_php_1 ... 
Starting opentelemetry-php_php_1 ... done
➜  opentelemetry-php git:(master) ✗ make examples
docker-compose run --rm php php ./examples/AlwaysOnTraceExample.php
➜  opentelemetry-php git:(master) ✗ 
```

I was then able to see the traces at http://localhost:9411/

Screenshot of this process attached.  I'm wondering if it's a MacOS thing?  Getting out my MacOS laptop to test this out as well.

![AlwaysOnTraceExample](https://user-images.githubusercontent.com/4099109/74400893-798b9480-4ded-11ea-9827-81c1ba1c2fa6.png)

**Comment by bobstrecansky** [comment]
Seeing the same thing on my MacOS laptop (Catalina):

```
23:20:11 bob@macos ~ opentelemetry-php master docker --version
Docker version 19.03.5, build 633a0ea
23:20:18 bob@macos ~ opentelemetry-php master git reset --hard upstream/master
HEAD is now at d8a554b Fixes Broken Example Code and Link in README (#59)
23:20:21 bob@macos ~ opentelemetry-php master docker-compose up -d
Starting opentelemetry-php_php_1 ...
Starting opentelemetry-php_php_1 ... done
23:20:25 bob@macos ~ opentelemetry-php master make examples
docker-compose run --rm php php ./examples/AlwaysOnTraceExample.php
```

<img width="1043" alt="image" src="https://user-images.githubusercontent.com/4099109/74401151-6a591680-4dee-11ea-8db8-4e7e46de4266.png">

**Comment by astorm** [comment]
Ah, thank you for the exercise.  That was useful.  I think I see my misunderstanding and I will say it out loud for the future folks googling around. Closing this out.

I had initially done a `docker-compose up`, tried to run the example programs _with my locally installed PHP_, and [ran into the errors/oversights I saw here](https://github.com/open-telemetry/opentelemetry-php/pull/59).  I also presumed the use of a `zipkin` hostname in this file was a similar oversight, so I changed it to `127.0.0.1` in order to get things running on my locally installed version of PHP.

**Then** I realized there was a `Makefile` file and a second container for PHP itself.  When I tried to run the docker container it failed because I still had the host in `./examples/AlwaysOnTraceExample.php` set to 127.0.0.1.  

The key thing I was missing/had-forgotten was docker-compose will automatically setup networking for each container it sets up.  So within the `php` container the `zipkin` hostname will resolve correctly. 

Thank you for the clarification, and apologies for the interruption.
